### ISSUE TITLE
Build error related to ferror function when cross compiling

### ISSUE THREAD

**Issue by gmarkall** [mageec/beebs#57]
First reported by @jeremybennett when building for RISC-V:

```
$ make V=1 2>&1 | tee /tmp/1.log
...
/bin/bash ../libtool  --tag=CC   --mode=compile riscv32-unknown-elf-gcc -DHAVE_CONFIG_H -I. -I../../../beebs/support -I..     -g -O2  -O2 -march=rv32imc -fdata-sections -ffunction-sections -Wl,-gc-sections  -MT dummy-libc.lo -MD -MP -MF .deps/dummy-libc.Tpo -c -o dummy-libc.lo ../../../beebs/support/dummy-libc.c
libtool: compile:  riscv32-unknown-elf-gcc -DHAVE_CONFIG_H -I. -I../../../beebs/support -I.. -g -O2 -O2 -march=rv32imc -fdata-sections -ffunction-sections -Wl,-gc-sections -MT dummy-libc.lo -MD -MP -MF .deps/dummy-libc.Tpo -c ../../../beebs/support/dummy-libc.c -o dummy-libc.o
In file included from ../../../beebs/support/dummy-libc.c:29:
../../../beebs/support/dummy-libc.c:248:1: error: expected identifier or '(' before 'int'
 ferror (FILE *stream __attribute__ ((unused)))
 ^~~~~~
../../../beebs/support/dummy-libc.c:248:1: error: expected ')' before '(' token
 ferror (FILE *stream __attribute__ ((unused)))
 ^~~~~~
Makefile:500: recipe for target 'dummy-libc.lo' failed
make[2]: *** [dummy-libc.lo] Error 1
```

and then by @LubomirBogdanov when building for the Taiyo Yuden EYSGJN board:

>I'm investigating a problem where to configure BEEBS I have to do

```
./configure CFLAGS=--nosys.specs --host=arm-none-eabi --with-chip=nrf51822 --with-board=ty_eysgjn
```
>
> but then when I try to make it, it fails as you've added a replacement (I think) for nosys.specs. Then I decided to remove this flag and add:
>
```
./configure CFLAGS=-nostdlib --host=arm-none-eabi --with-chip=nrf51822 --with-board=ty_eysgjn
```
>
> which configures and builds successfully, but the linking now is wrong and I get empty object files. I will try to see what's the problem further.
>
> However one thing is sure - I always get:
>
```
dummy-libc.c:248:1: error: expected identifier or '(' before 'int'
 ferror (FILE *stream __attribute__ ((unused)))
 ^~~~~~
dummy-libc.c:248:1: error: expected ')' before '(' token
 ferror (FILE *stream __attribute__ ((unused)))
```
>
> until I comment this empty function out. This looks mysterious as all the other functions, having the same attribute parameters, pass.

**Comment by gmarkall** [comment]
Fix proposed in #58

**Comment by gmarkall** [comment]
Fixed in #58.
